### DISCUSSION TITLE
Custom scripts not updated

### DISCUSSION THREAD

**Discussion by nyevess** [netbox-community/netbox#14428] [Help Wanted!]
Hi community, I'm triying to add new custom scripts to Netbox, but didn't appear in the web, there are new steps to make this operation?. I have 4 scripts in the folder:
But I only can see 2, could you help me? Thanks! 
![image](https://github.com/netbox-community/netbox/assets/69692460/8eaaff98-a612-4eb1-8088-d46cff74e341)
![image](https://github.com/netbox-community/netbox/assets/69692460/37227d41-589b-45da-a348-a65ddb261b09)

**Comment by deku-m** [comment]
Did you also try to upload them through the webgui?

**Comment by Lon1** [comment]
Hi @nyevess - What is your NetBox environment? What version of code? How can you reproduce the issue?  Are you copying in custom scripts in your startup or adding them via the UI?  Also, looks like you haven't changed the owner of the scripts/reports/media dirs as described in the docs: https://docs.netbox.dev/en/stable/installation/3-netbox/#create-the-netbox-system-user

**Comment by candlerb** [comment]
> Hi community, I'm triying to add new custom scripts to Netbox, but didn't appear in the web, there are new steps to make this operation?

Yes, with recent versions of Netbox you must upload them via the web UI (which also registers them in the database). Simply dropping them into the scripts folder is no longer sufficient.

I think this probably happened in v3.5.0, when remote data sources were added.

**Reply by nyevess** [reply]
Hello, i've tried to add from GUI but I have the error below:
![image](https://github.com/netbox-community/netbox/assets/69692460/0af3944a-c883-41eb-bdac-f8db98e77283)

**Reply by nyevess** [reply]
My netbox version is 3.6.1

**Reply by candlerb** [reply]
You have an error in your script - or possibly a permissions problem or maybe you removed it without telling the GUI. Unfortunately, Netbox doesn't show you the actual problem (#12766). If the problem is an error in your script, that issue shows you a workaround way to debug it manually.

Alternatively, if you just want to get working again, move the script out of the way (if it exists) and replace it with a zero byte file.

```
cd /opt/netbox/netbox/scripts
mv move_decoders.py move_decoders.py.broken
touch move_decoders.py
```

**Reply by nyevess** [reply]
Thank you for your response, trying that you said, I think there is a problem in my script. I have to recheck the code. Thanks!

**Reply by cmcknz77** [reply]
Worth your while checking the permissions and ownership of the script file too by navigating to the scripts folder 
/opt/netbox/netbox/scripts and issuing the command '**ls -lhart**'

Each .py script file should be owned by the user account that netbox runs with (in my case 'netbox': **chown netbox:netbox script_filename.py**).
Each file also needs to be correctly permissioned with chmod (but I don't know what the correct permissions need to be).
